frontend/apps/filemanager/filemanagerutil.lua

```diff
@@ -138,34 +130,66 @@ end
             local confirmbox = ConfirmBox:new{
                 text = T(_("Reset this document?") .. "\n\n%1\n\n" ..
-                    _("Document progress, settings, bookmarks, highlights, notes and custom cover image will be permanently lost."),
+                         _("Resetted information will be permanently lost."),
                     BD.filepath(file)),
                 ok_text = _("Reset"),
                 ok_callback = function()
-                    local custom_metadata_purged = filemanagerutil.purgeSettings(file)
-                    if custom_metadata_purged then -- refresh coverbrowser cached book info
-                        local FileManager = require("apps/filemanager/filemanager")
-                        local ui = FileManager.instance
-                        if not ui then
-                            local ReaderUI = require("apps/reader/readerui")
-                            ui = ReaderUI.instance
-                        end
+                    local data_to_purge = {
+                        doc_settings         = check_button_settings.checked,
+                        custom_cover_file    = check_button_cover.checked and custom_cover_file,
+                        custom_metadata_file = check_button_metadata.checked and custom_metadata_file,
+                    }
+                    DocSettings:open(file):purge(nil, data_to_purge)
+                    if data_to_purge.custom_cover_file or data_to_purge.custom_metadata_file then
+                        local ui = require("apps/filemanager/filemanager").instance
+                                or require("apps/reader/readerui").instance
                         if ui and ui.coverbrowser then
-                            ui.coverbrowser:deleteBookInfo(file)
+                            ui.coverbrowser:deleteBookInfo(file) -- refresh coverbrowser cached book info
                         end
                     end
-                    require("readhistory"):fileSettingsPurged(file)
+                    if data_to_purge.doc_settings then
+                        require("readhistory"):fileSettingsPurged(file)
+                    end
                     caller_callback()
                 end,
             }
+            check_button_settings = CheckButton:new{
+                text = _("document settings, progress, bookmarks, highlights, notes"),
+                checked = has_sidecar_file,
+                enabled = has_sidecar_file,
+                parent = confirmbox,
+            }
+            confirmbox:addWidget(check_button_settings)
+            check_button_cover = CheckButton:new{
+                text = _("custom cover image"),
+                checked = has_custom_cover_file,
+                enabled = has_custom_cover_file,
+                parent = confirmbox,
+            }
+            confirmbox:addWidget(check_button_cover)
+            check_button_metadata = CheckButton:new{
+                text = _("custom book properties"),
```

Comment:
We handlle book metadata as `book_props` internally.
Dunno if we should use "book properties" or "book metadata" in the UI. "book property" feels more vague than metadata to me.
(Also in the previous PR allowing editing, we show "Book property: title" and "Edit book property: Title".)

Reply:
Another "Metadata" is explained here
https://github.com/koreader/koreader/blob/a767ad44db5a22c6d14c505241e8698fa3113c69/frontend/ui/elements/common_settings_menu_table.lua#L547

Reply:
Amazon says "metadata fields":
https://kdp.amazon.com/en_US/help/topic/G201097560

Reply:
I guess we felt the need to mention "metadata" in "(collectively known as metadata)" because our docsettings ("docsettings.epub.lua" would have been a quite good name) file is named `metadata.epub.lua`, as a way to indicate to the user who has met them in some file browser that this file is what we are talking about.
Too late to rename all these `metadata.epub.lua` :) (and external tools expect them named liked that) but I would have called that doc settings or doc properties.

And I would, as Amazon, name the book metadata fields... book metadata :)

The simpler would be to remove "(collectively known as metadata)" from that string. Do we mention "metadata" elsewhere?
Or find another user facing name for these title/authors/language/keywords...
Dunno.
Needs more input.

Reply:
I vaguely recall some discussion about that string, but not why "collectively known as metadata" is in there.